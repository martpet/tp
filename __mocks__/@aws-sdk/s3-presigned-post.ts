export const createPresignedPost = vi.fn().mockResolvedValue('dummyPresignedPost');
