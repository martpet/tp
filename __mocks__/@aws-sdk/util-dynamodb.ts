export const unmarshall = vi
  .fn()
  .mockName('unmarshall')
  .mockReturnValue('dummyUnmarshalledItems');
