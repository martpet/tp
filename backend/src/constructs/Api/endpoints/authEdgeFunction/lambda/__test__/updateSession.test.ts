import { DynamoDBDocumentClient, UpdateCommand } from '@aws-sdk/lib-dynamodb';
import { mockClient } from 'aws-sdk-client-mock';

import { updateSession } from '../updateSession';

const ddbMock = mockClient(DynamoDBDocumentClient);

const args = [
  {
    sessionId: 'dummySessoinid',
    idToken: 'dummyIdToken',
  },
] as Parameters<typeof updateSession>;

beforeEach(() => {
  ddbMock.reset();

  ddbMock.on(UpdateCommand).resolves({
    Attributes: {
      dummyUpdateCommandAttr: 'dummyUpdateCommandAttrValue',
    },
  });
});

describe('updateSession', () => {
  it('sends "UpdateCommand" to DynamoDB with correct args', async () => {
    await updateSession(...args);
    expect(ddbMock.commandCalls(UpdateCommand)[0].args[0].input).toMatchSnapshot();
  });

  it('resolves with a correct value', () => {
    return expect(updateSession(...args)).resolves.toMatchSnapshot();
  });
});
