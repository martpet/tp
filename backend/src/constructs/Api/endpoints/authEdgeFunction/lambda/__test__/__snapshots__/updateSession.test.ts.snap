// Vitest Snapshot v1

exports[`updateSession > resolves with a correct value 1`] = `
{
  "Attributes": {
    "dummyUpdateCommandAttr": "dummyUpdateCommandAttrValue",
  },
}
`;

exports[`updateSession > sends "UpdateCommand" to DynamoDB with correct args 1`] = `
{
  "ExpressionAttributeNames": {
    "#idToken": "idToken",
  },
  "ExpressionAttributeValues": {
    ":idToken": "dummyIdToken",
  },
  "Key": {
    "id": "dummySessoinid",
  },
  "TableName": "TripPics-Sessions",
  "UpdateExpression": "set #idToken = :idToken",
}
`;
